server/service/src/sensor/berlinger.rs

```diff
@@ -0,0 +1,380 @@
+        }
+        let new_sensor = SensorRow {
+            id: uuid(),
+            serial: temperature_sensor.serial.clone(),
+            name: temperature_sensor.name.clone(),
+            store_id: Some(store_id.to_string()),
+            location_id: None,
+            last_connection_datetime: None,
+            battery_level: None,
+            is_active: true,
+            log_interval: interval_seconds,
+            r#type: SensorType::Berlinger,
+        };
+        SensorRowRepository::new(connection).upsert_one(&new_sensor)?;
+        println!("Added sensor {:?} ", new_sensor);
+        Ok(())
+    }
+}
+
+pub fn read_sensors(
+    connection: &StorageConnection,
+    store_id: &str,
+) -> Result<Vec<String>, SingleRecordError> {
+    let mut sensors_processed: Vec<String> = Vec::new();
+
+    let sensor_serials =
+        read_connected_serials().map_err(|err| SingleRecordError::NotFound(err))?;
+    let expected_sensor_count = sensor_serials.len();
+
+    for current_serial in sensor_serials {
+        let mut temperature_sensor = temperature_sensor::read_sensor(&current_serial, true)
+            .map_err(|err| SingleRecordError::NotFound(err))?;
+        sensor_add_if_new(connection, &store_id, &temperature_sensor)?;
+
+        let result = get_matching_sensor_serial(connection, &current_serial)?;
+
+        if let Some(mut record) = result.clone().pop() {
+            let sensor_id = &record.sensor_row.id;
+            // Filter sensor data by previous last connected time
+            let last_connected = record.sensor_row.last_connection_datetime;
+            temperature_sensor =
+                temperature_sensor::filter_sensor(temperature_sensor, last_connected, None, true);
+
+            let mut temperature_breach_configs =
+                get_breach_configs_for_sensor(connection, sensor_id)?;
+            println!(
+                "{} temperature breach configs before",
+                temperature_breach_configs.len()
+            );
+
+            if let Some(temperature_sensor_configs) = &temperature_sensor.configs {
+                for temperature_sensor_config in temperature_sensor_configs {
+                    sensor_add_breach_config_if_new(
+                        connection,
+                        &record.sensor_row,
+                        temperature_sensor_config,
+                    )?;
+                }
+            }
+
+            temperature_breach_configs = get_breach_configs_for_sensor(connection, sensor_id)?;
+            println!(
+                "{} temperature breach configs after",
+                temperature_breach_configs.len()
+            );
+
+            let mut temperature_breaches = get_breaches_for_sensor(connection, sensor_id)?;
+            println!("{} temperature breaches before", temperature_breaches.len());
+
+            if let Some(temperature_sensor_breaches) = &temperature_sensor.breaches {
```

Comment:
can do temperature_sensor.breaches.unwrap_or_default(), also to reduce indentation